packages/babel-traverse/src/path/context.ts

```diff
@@ -188,9 +189,9 @@ export function setContext<S = unknown>(
 export function resync(this: NodePath) {
   if (this.removed) return;
 
-  this._resyncParent();
-  this._resyncList();
-  this._resyncKey();
+  _resyncParent.call(this);
```

Comment:
What do you think of changing all these "private methods" to accept the `NodePath` as the first parameter instead of this, so that we can do `_resyncParent(this)` instead of `_resyncParent.call(this)`?

Then, in index.ts we have to do
```js
Object.assign(NodePath_Final.prototype, {
  ...
  _resyncParent: thisify(NodePath_context._resyncParent)
  ...
```

where `thisify` is
```js
function thisify(fn) {
  return function() {
    return fn(this, ...arguments);
  }
}
```

in practice those thisified functions will never be called anyway, since they are only called internally by `@babel/traverse`.

Reply:
Maybe we can do this after Babel 8 is more stable and avoid us needing to expose certain methods.